pkg/cmd/roachtest/cluster.go

```diff
@@ -1936,6 +1936,7 @@ func (c *clusterImpl) StartE(
 	// Set the same seed on every node, to be used by builds with
 	// runtime assertions enabled.
 	setUnlessExists("COCKROACH_RANDOM_SEED", c.cockroachRandomSeed())
+	l.Printf("ROACHTEST COCKROACH_RANDOM_SEED: %d", c.cockroachRandomSeed())
```

Comment:
Two things to consider about this approach:

* It will be logged even if we are not using a metamorphic build.
* It will be logged every time we start cockroach, which could be several times in a test (especially in upgrade tests).

Maybe the best place to put this type of message is during cluster setup (only once); it could even be done as part of your work in #113301, where there's a pretty well-defined moment where we decide if we are running with metamorphic builds or not (and that only happens once during a test's lifetime).

Thoughts?

*****

Nit: I would also slightly change the messaging, something like `to reproduce failures using the same metamorphic constants, run this test with %s=%d` (roachtest env var, seed). Doesn't have to be this exactly, but it would be nice to convey that information in the log message.

Reply:
Sounds good to me